src/nbl/video/COpenGLFunctionTable.h

```diff
@@ -257,11 +266,27 @@ class COpenGLFunctiontable : public COpenGL_FunctionTableBase
 		gl4Query(_egl),
 		gl4General(_egl),
 		gl4Sync(_egl),
+		gl4Debug(_egl),
         features(_features)
     {
 
     }
 
+	void extGlDebugMessageControl(GLenum source, GLenum type, GLenum severity, GLsizei count, const GLuint* ids, GLboolean enabled) override
+	{
+		if (gl4Debug.pglDebugMessageCallback)
+			gl4Debug.pglDebugMessageCallback(source, type, severity, count, ids, enabled);
+		else if (gl4Debug.pglDebugMessageCallbackARB)
+			gl4Debug.pglDebugMessageCallbackARB(source, type, severity, count, ids, enabled);
+	}
+	void extGlDebugMessageCallback(GLDebugCallbackType callback, const void* userParam) override
+	{
+		if (gl4Debug.pglDebugMessageCallback)
+			gl4Debug.pglDebugMessageCallback(callback, userParam);
+		else if (gl4Debug.pglDebugMessageCallbackARB)
+			gl4Debug.pglDebugMessageCallbackARB(callback, userParam);
+	}
```

Comment:
btw there's a thing somewhere in the Vulkan validation layers (there might be also a vulkan capability/extension for debug messages from the driver) which allows for a similar thing

Would be nice if the Logical device (or instance or whatever) had a uniform method for registering a callback with a uniform signature to log the debug messages from the driver, because a lot of examples have hacky code for the debug message callbacks (using opengl function pointers directly to set up the debug callbacks)